src/util/status_file_test.go

```
@@ -0,0 +1,186 @@
+		It("should wait until status changes", func() {
+			statusFilePath := filepath.Join(tempDir, "status.json")
+
+			// Create empty file
+			err := os.WriteFile(statusFilePath, []byte(""), 0644)
+			Expect(err).ToNot(HaveOccurred())
+
+			go func() {
+				time.Sleep(20 * time.Millisecond)
+				content := StatusFileContent{Status: StatusWaitingForAgent}
+				data, _ := json.Marshal(content)
+				os.WriteFile(statusFilePath, data, 0644)
```

golangci-lint (2.12.2)

[error] Error return value of `os.WriteFile` is not checked

(errcheck)